lib/inspector/inspector.dart

```diff
@@ -0,0 +1,181 @@
+            ..add(<CoreElement>[
+              ServiceExtensionButton(extensions.performanceOverlay).button,
+              ServiceExtensionButton(extensions.togglePlatformMode).button,
+              ServiceExtensionButton(extensions.debugPaint).button,
+              ServiceExtensionButton(extensions.debugPaintBaselines).button,
+              ServiceExtensionButton(extensions.slowAnimations).button,
+
+              // These extensions could be demoted to a menu.
+              ServiceExtensionButton(extensions.repaintRainbow).button,
+              ServiceExtensionButton(extensions.debugAllowBanner).button,
+            ]),
+        ]),
+      inspectorContainer = div(c: 'inspector-container'),
+    ]);
+
+    serviceManager.onConnectionAvailable.listen(_handleConnectionStart);
+    if (serviceManager.hasConnection) {
+      _handleConnectionStart(serviceManager.service);
+    }
+    serviceManager.onConnectionClosed.listen(_handleConnectionStop);
+  }
+
+  void _handleConnectionStart(VmService service) async {
+    refreshTreeButton.disabled = false;
+
+    final Spinner spinner = Spinner()..clazz('padded');
+    inspectorContainer.add(spinner);
+
+    try {
+      inspectorService = await InspectorService.create(service);
+      final pubRootDirectory =
+          await inspectorService.inferPubRootDirectoryIfNeeded();
+      if (pubRootDirectory != null) {
+        // TODO(jacobr): add ui to view and set a different pub root directory and
+        // display the pub root directory in the status bar.
+      }
+    } finally {
+      spinner.element.remove();
+      refreshTreeButton.disabled = false;
+    }
+
+    // TODO(jacobr): support the Render tree, Layer tree, and Semantic trees as
+    // well as the widget tree.
+
+    inspectorController = InspectorController(
+      inspectorTreeFactory: ({
+        summaryTree,
+        treeType,
+        onSelectionChange,
+        onExpand,
+        onHover,
+      }) {
+        if (_useHtmlInspectorTreeRenderer) {
+          return InspectorTreeHtml(
+            summaryTree: summaryTree,
+            treeType: treeType,
+            onSelectionChange: onSelectionChange,
+            onExpand: onExpand,
+            onHover: onHover,
+          );
+        }
+        return InspectorTreeCanvas(
+          summaryTree: summaryTree,
+          treeType: treeType,
+          onSelectionChange: onSelectionChange,
+          onExpand: onExpand,
+          onHover: onHover,
+        );
+      },
+      inspectorService: inspectorService,
+      treeType: FlutterTreeType.widget,
+    );
+    final InspectorTreeWeb inspectorTree = inspectorController.inspectorTree;
+    final InspectorTreeWeb detailsInspectorTree =
+        inspectorController.details.inspectorTree;
+
+    final elements = [
+      inspectorTree.element.element,
+      detailsInspectorTree.element.element
+    ];
+    inspectorContainer.add(elements);
+    splitterSubscription = flexSplitBidirectional(elements);
```

Comment:
I would use `defaultSplitterWidth` for the splitter width. I see that this version of the method doesn't take preferred ratios for the child sizes, but if we can add it you may try out a 40/60 or 33/66 split (see if giving more weight to the RHS looks good).

Reply:
added default size options for the bidirectional splitter which is a bit tricky as it can be both horizontal and vertical depending on the window size.
Agreed that a 60/40 split looks much better. Use 65/35 split in the other direction in the vertical layout case as in that case the details panel should be smaller not larger.